### PR TITLE
feat: add property to accessible BrowserWindow window title

### PR DESCRIPTION
**Pull request by mshoho**
#### Description of Change

Sometimes it's necessary to convey more information about the window to the screen reader users only (simply putting everything to the window title might be unnecessarily noisy).

For example, Chromium uses that technique to tell the screen reader users that the window is in incognito mode (the incognito window looks differently and doesn't have «incognito» in the title, but for the blind users the screen reader will announce that it's incognito).

Chromium has GetAccessibleWindowTitle() method internally. This PR exposes a way to customize the accessible title via Electron API.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [ ] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [x] relevant documentation is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Added `win.accessibleTitle property` to be able to augment the window title for the screen readers.

### CODE REVIEW

**Comment by alexeykuzmin** [comment]
@mshoho please use the PR template for description.

File: shell/browser/native_window.h [hunk 1]
```diff
@@ -132,6 +132,12 @@ class NativeWindow : public base::SupportsUserData,
   virtual void Invalidate() = 0;
   virtual void SetTitle(const std::string& title) = 0;
   virtual std::string GetTitle() = 0;
+
+  // Ability to augment the window title for the screen readers.
+  base::string16 GetAccessibleWindowTitle() const override;
```
Comment (miniak) [inline shell/browser/native_window.h]: please move to the section with other `WidgetDelegate` overrides

File: shell/browser/native_window.cc [hunk 2]
```diff
@@ -582,6 +582,22 @@ const views::Widget* NativeWindow::GetWidget() const {
   return widget();
 }
 
+base::string16 NativeWindow::GetAccessibleWindowTitle() const {
```
Comment (miniak) [inline shell/browser/native_window.cc]: please move to the section with other `WidgetDelegate` overrides

File: docs/api/browser-window.md [hunk 3]
```diff
@@ -1227,6 +1227,16 @@ Returns `String` - The title of the native window.
 **Note:** The title of the web page can be different from the title of the native
 window.
 
+#### `win.setAccessibleTitle(title)`
```
Comment (miniak) [inline docs/api/browser-window.md]: please add to `docs/api/modernization/property-updates.md`

**Comment by miniak** [comment]
please use the correct PR template

File: docs/api/browser-window.md (hunk 3 above)
Comment (codebytere) [reply docs/api/browser-window.md]: this is new, so it doesn't need to go there!

Comment (MarshallOfSound) [reply docs/api/browser-window.md]: That said this should definitely be a property `win.accessibleTitle`

File: shell/browser/api/atom_api_top_level_window.cc [hunk 4]
```diff
@@ -1108,6 +1116,8 @@ void TopLevelWindow::BuildPrototype(v8::Isolate* isolate,
       .SetMethod("getPosition", &TopLevelWindow::GetPosition)
       .SetMethod("setTitle", &TopLevelWindow::SetTitle)
       .SetMethod("getTitle", &TopLevelWindow::GetTitle)
+      .SetMethod("setAccessibleTitle", &TopLevelWindow::SetAccessibleTitle)
```
Comment (MarshallOfSound) [inline shell/browser/api/atom_api_top_level_window.cc]: This should be a property `accessibleTitle` with a setter/getter 👍

File: docs/api/browser-window.md [hunk 5]
```diff
@@ -379,6 +379,8 @@ It creates a new `BrowserWindow` with native properties as set by the `options`.
     * `disableHtmlFullscreenWindowResize` Boolean (optional) - Whether to
       prevent the window from resizing when entering HTML Fullscreen. Default
       is `false`.
+    * `accessibleTitle` String (optional) - a part of Window title visible for
```
Comment (miniak) [inline docs/api/browser-window.md]: I think it should be visible to instead of for

**Comment by miniak** [comment]
@MarshallOfSound can you please review the current version?

File: shell/browser/native_window.cc [hunk 6]
```diff
@@ -582,6 +582,22 @@ const views::Widget* NativeWindow::GetWidget() const {
...
+  if (accessible_title_.empty()) {
+    return views::WidgetDelegate::GetAccessibleWindowTitle();
+  }
+
+  return base::UTF8ToUTF16(accessible_title_);
```
Comment (MarshallOfSound) [inline shell/browser/native_window.cc]: Is there a reason we can't store this as `base::string16`.  Storing as `std::string` just to convert to `string16` every time _something_ in chrome calls this getter seems a bit rough

File: docs/api/browser-window.md (hunk 5 above)
Comment (MarshallOfSound) [inline docs/api/browser-window.md]: `String (optional) - An alternative title string provided only to accessibility tools such as screen readers.  This string is not directly visible to users.`

File: shell/browser/native_window.cc (hunk 6 above)
Comment (mshoho) [reply shell/browser/native_window.cc]: The only reason is that I believe I've seen something similar somewhere else. I've switched to storing it as `base::string16`.

File: shell/browser/native_window.cc (hunk 2 above)
[the remark this answers is not in the document]
Comment (mshoho) [reply shell/browser/native_window.cc]: It's a Chromium's method and Chromium gives the ability to override it. And Chromium calls it internally constructing the title for the screen readers.

[the remark this answers is not in the document]
Comment (mshoho) [reply shell/browser/native_window.cc]: More precisely it's a method of Chromium's WidgetDelegate and I'm just overriding it.

**Pull request merged**